src/navigator_data_ingest/base/updated_document_actions.py

```diff
@@ -271,11 +271,55 @@ def parse(
     return [error for error in errors if error is not None]
 
 
+def update_field_in_all_occurences(
+    update: Tuple[str, Update],
+    update_config: UpdateConfig,
+) -> List[Union[str, None]]:
+    """Update the document slug in all occurences of the document in s3."""
+    # TODO Do we need to archive on slug updates? The reason for this was expensive
+    # translation and text extraction costs not slug updates?
+    # TODO This can be made more generic.
+
+    document_id, document_update = update
```

Comment:
Needing this line feels like the update object isn't giving us everything we want. Could the id become a property instead?

Reply:
That is a very good point, I will ticket and add to the backlog as that will require another PR to the data access library!